test/e2e/operator_test.go

```diff
@@ -1,26 +1,316 @@
 package e2e
 
 import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
 	"testing"
+	"time"
 
+	corev1 "k8s.io/api/core/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	kapierrs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
+	policyclientv1beta1 "k8s.io/client-go/kubernetes/typed/policy/v1beta1"
 
 	"github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/operatorclient"
 	test "github.com/openshift/cluster-kube-controller-manager-operator/test/library"
+	"github.com/prometheus/common/model"
 )
 
-func TestOperatorNamespace(t *testing.T) {
+const (
+	kcmoNamespace        = "openshift-kube-controller-manager-operator"
+	kcmoSigningKeySecret = "kube-controller-manager-operator-serving-cert"
+)
+
+func TestE2E(t *testing.T) {
+	// See https://bugzilla.redhat.com/show_bug.cgi?id=1762888
 	kubeConfig, err := test.NewClientConfigForTest()
 	if err != nil {
 		t.Fatal(err)
 	}
+
 	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
 	if err != nil {
 		t.Fatal(err)
 	}
-	_, err = kubeClient.CoreV1().Namespaces().Get(operatorclient.OperatorNamespace, metav1.GetOptions{})
+
+	policyClient, err := policyclientv1beta1.NewForConfig(kubeConfig)
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	// test that operator namespace exists
+	t.Run("OperatorNamespace", func(t *testing.T) {
```

Comment:
Is the operator namespace being present a precondition for the other test to execute? Since `t.Run` isolates failure, enclosing this check would result in suite failure but would not prevent `PodDisruptionBudgetAtLimit alert` from executing.

Reply:
Good point, I removed the nesting.